src/codegen/compile.ml

```diff
@@ -1465,19 +1474,32 @@ module BitTagged = struct
 end (* BitTagged *)
 
 module Tagged = struct
-  (* Tagged objects have, well, a tag to describe their runtime type.
-     This tag is used to traverse the heap (serialization, GC), but also
-     for objectification of arrays.
+  (* Tagged objects all have an object header consisting of a tag and a forwarding pointer.
+     The tag is to describe their runtime type and serves to traverse the heap 
+     (serialization, GC), but also for objectification of arrays.
 
      The tag is a word at the beginning of the object.
 
+     They forwarding pointer supports object moving in the incremental garbage collection.
+
+         obj header
+     ┌──────┬─────────┬──
+     │ tag  │ fwd ptr │ ...
+     └──────┴─────────┴──
+
      All tagged heap objects have a size of at least two words
```

Comment:
```suggestion
     All tagged heap objects have a size of at least three words
```
?

Reply:
Thank you, I indeed should refine this comment. There is a special exception with `NULL` that now has a size of two words and even occupied only one word before this PR. However, as it resides in the static heap, there is no problem with the indirection during the copying GC.